test/build.info

```diff
@@ -593,11 +593,19 @@ ENDIF
        next if grep { lc("$name.h") =~ m/$_/i } @nogo_headers_re;
        $OUT .= <<"_____";
 
-  PROGRAMS{noinst}=buildtest_$name
+  PROGRAMS{noinst}=buildtest_c_$name
+  SOURCE[buildtest_c_$name]=buildtest_$name.c
   GENERATE[buildtest_$name.c]=generate_buildtest.pl $name
-  SOURCE[buildtest_$name]=buildtest_$name.c
-  INCLUDE[buildtest_$name]=../include
-  DEPEND[buildtest_$name]=../libssl ../libcrypto
+  INCLUDE[buildtest_c_$name]=../include
+  DEPEND[buildtest_c_$name]=../libssl ../libcrypto
+_____
+       $OUT .= <<"_____" if $config{CXX};
+
+  PROGRAMS{noinst}=buildtest_cc_$name
+  SOURCE[buildtest_cc_$name]=buildtest_$name.cc
+  GENERATE[buildtest_$name.cc]=generate_buildtest.pl $name
```

Comment:
Is there a particular reason why you create extra `buildtest_*.cc` files, instead of just using the `buildtest_*.c` files? I ask, because currently the file contents are identical:

```
for f in $(find test -name '*.cc'); do echo diff $f $(dirname $f)/$(basename $f .cc).c; done
```

Reply:
Ups, I had to remove an `echo` from the previous command.  ;-)

Reply:
Ah forget my last question, it was stupid. Of course the compiler gets selected by the file extension.

Reply:
I was considering using attributes to sway the compiler choice, but that's a much bigger effort, wasn't sure it was worth it for such small files.

Reply:
Does this mean A C++ compiler is required to do 'make test' now?

Reply:
No, it means that extra build test will be performed *if* a C++ compiler has been configured.